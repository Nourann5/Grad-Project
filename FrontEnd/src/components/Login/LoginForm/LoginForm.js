import React, { useEffect, useRef, useState } from 'react'
import {ReactComponent as FacebookIcon} from 'assets/icons/facebook.svg'
import {ReactComponent as GoogleIcon} from 'assets/icons/google.svg'
import {ReactComponent as LinkedInIcon} from 'assets/icons/linkedin.svg'
import {ReactComponent as PasswordViewed} from 'assets/icons/passwordViewed.svg'
import styles from './LoginForm.module.css'
import { NavLink, useNavigate } from 'react-router-dom'
import { useForm } from 'react-hook-form'
import { toast } from 'react-toastify'
import ButtonsLoading from 'components/Global/Elements/ButtonsLoading/ButtonsLoading'
import { axiosConfig } from 'utils/axiosConfig'
import { emailPattern } from 'utils/features'
import { useDispatch, useSelector } from 'react-redux'
import Cookies from 'js-cookie'
import { loggingIn } from 'store/Login/LoginActions'
import jwtDecode from 'jwt-decode'
function LoginForm() {
    const {register ,getValues,handleSubmit ,formState:{errors}} = useForm({validate:'onChange'})
    const navigate = useNavigate()
    const passwrodInput = useRef(null)
    const googleRef = useRef(null)

    const [isPasswordVisible , setIsPasswordVisible] = useState(false)
    const [isSubmitting , setIsSubmitting]=useState(false)
    const userr=  useSelector(state=>state.LoginReducer.user)
    const dispatch = useDispatch()

    function showHidePassword(){
        let passIcon = document.getElementById('Path_3314')
        setIsPasswordVisible((prevState)=>!prevState)
        passIcon.classList.toggle(styles['toggle-view-pass'])
    }

    let submitForm =(data)=>{
            let formData =new FormData()
            Object.keys(getValues()).forEach(key=>{
                formData.append(`${key}`,getValues()[key])
            })
            setIsSubmitting(true)
            axiosConfig.post('user/login-user',formData,{
        }).then(res=>{
            toast.success('User Logged Successfully')
            setIsSubmitting(false)
            console.log(res.data)
            Cookies.set('token',res?.data?.data?.token)
            Cookies.set('user',JSON.stringify(res?.data?.data?.user))
            // localStorage.setItem('token',res.data.token)
            Cookies.set('permissions',JSON.stringify(res.data.permission))
            dispatch(loggingIn(res.data))
            navigate('/portal')
        }).catch(err=>{
            toast.error(err?.response?.data?.message||'Something went wrong')
            setIsSubmitting(false)
        })
    }

    // function signInWithGoogle(){
    //     console.log('adsdasdsa',googleRef.current,googleRef.current.querySelector('iframe').querySelector('.nsm7Bb-HzV7m-LgbsSe'))
    //     googleRef.current.querySelector('iframe').click()
    // }

    // function handleCallBackResponse(response){
    //     console.log('dasdsdas',response.credential)
    //     let userObject = jwtDecode(response.credential)
    //     console.log('userObjectuserObject',userObject)
    // }

    // useEffect(()=>{
    // /* global google*/
    //     google.accounts.id.initialize({
    //         client_id:'829484648717-dgvvef2h3q23aslobuo4h993kagt1333.apps.googleusercontent.com',
    //         callback:handleCallBackResponse
    //     })
    //     google.accounts.id.renderButton(
    //         document.getElementById('googleCont'),
    //         {theme:'outline',size:'large'}
    //     ) 
    // },[])

    return (
        <>
            <div className={styles["login-form-content"]}>
                <div className={styles["login__form-content-container"]}>
                    <p className={styles["login-form-title"]}>Welcome Back</p>
                    <form className={styles["login-form"]} onSubmit={handleSubmit(submitForm)}>
                        <div className={styles["form__input-cont"]}>
                            <label htmlFor="form__input-email" className={styles["form__input-label"]}>Email</label>
                            <input 
                                type="email" 
                                className={`${styles["form__input"]} ${errors?.email ?'modal__form-input--error':''}`} 
                                id="form__input-email" 
                                placeholder="Type Email Here"
                                {...register('email',{required:'Email is required',
                                    pattern:{
                                        value:emailPattern,
                                        message:'Email must be like invo***@academy**.***'
                                    }})}
                            />
                            {errors?.email &&<span className={'modal__form-input-error-message'}>{errors.email?.message}</span>}
                        </div>
                        <div className={styles["form__input-cont"]}>
                            <label htmlFor="form__input-password" className={styles["form__input-label"]}>Password</label>
                            <div className={styles["form__input-password"]}>
                                <input 
                                    type={`${isPasswordVisible?'text':'password'}`} 
                                    className={`${styles["form__input"]} ${errors?.password ?'modal__form-input--error':''}`} 
                                    ref={passwrodInput}
                                    id="form__input-password" 
                                    placeholder='Type Password Here'
                                    {...register('password',{required:'Password is required'})}
                                />
                                {errors?.password &&<span className={'modal__form-input-error-message'}>{errors.password?.message}</span>}
                                <button type="button" onClick={showHidePassword} className={styles["form__view-pass"]}>
                                    <PasswordViewed className={styles["form__view-pass-icon"]}/>
                                </button>
                            </div>
                        </div>
                        <div className={styles["login__remeber-me"]}>
                            <div className={styles["login__remeber-me-cont"]}>
                                <input type="checkbox" id="login__remeber-me-check" className={styles["login__remeber-me-check"]}/>
                                <label htmlFor="login__remeber-me-check" className={styles["login__remeber-me-label"]}>Remember Me</label>
                            </div>
                            <NavLink to='/forget' className={styles["login__forget-password"]}>Forgot Password?</NavLink>
                        </div>
                        <button type="submit" className={styles["form__submit"]}>{isSubmitting?<ButtonsLoading/>:'Login'}</button>
                    </form>
                    {/* <p className={styles["login__with-other-ways"]}>
                        Or Login With
                    </p>
                    <div className={styles["login__social-media-cont"]}>
                        <button className={styles["login__social-media-button"]}>
                            <FacebookIcon/>
                        </button>
                        <button className={styles["login__social-media-button"]}>
                            <GoogleIcon/>
                        </button>
                        <button className={styles["login__social-media-button"]}>
                            <LinkedInIcon/>
                        </button>
                    </div> */}
                </div>
            </div>
            <p className={styles["login__dont-have-accout"]}>
                Don't Have An Account ? 
                <NavLink to='/register' className={styles["login__move-to-register"]}> Register</NavLink>
            </p>
        </>
    )
}

export default LoginForm