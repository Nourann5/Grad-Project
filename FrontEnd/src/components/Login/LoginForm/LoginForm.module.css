.login__form-head{
    display: flex;
    align-items: center;
    margin-bottom: 15px;
}
.sign-up__logo-cont{
    display: flex;
    flex-direction:column;
    align-items: center;
    justify-content: center;
    margin: auto;
}
.sign-up__logo{
    margin-bottom: 8px;
}
.sign-up__logo-role{
    color:var(--text-color-dark);
    font-family: "PoppinsSemiBoldItalic";
    font-size:20px;
    margin: 0;
}
.login__return-para{
    position: absolute;
    color:var(--text-color-dark);
    font-size:18px;
    font-family:'PoppinsMedium';
    transition:var(--transition);
    display: flex;
    align-items: center;
}
.login__return-para:hover{
    color:var(--main-color);
}
.login__return-icon{
    margin-right: 5px;
    width: 20px;
    height: 20px;
}
.login__logo{
    margin: auto;
    width: 70px;
    height: 70px;
}
.login-form-content{
    width:50%;
    margin: auto;
    background-color:white;
    box-shadow: 10px 10px 35px #4807E61A;
    padding: 30px 0;
    margin-bottom: 10px;
    border-radius: 5px;
}
.login__form-content-container{
    width:80%;
    margin: auto;

}
.login-form-title{
    font-size: 26px;
    font-family: 'PoppinsSemiBold';
    color: var(--text-color-dark);
    margin-bottom: 10px;
    text-align: center;
}
.form__input-cont{
    margin-bottom: 15px;
    display: flex;
    flex-direction: column;
}
.form__input-label{
    color:var(--main-color);
    font-family:'PoppinsMedium';
    font-size: 16px;
    margin-bottom: 8px;
}
.form__input{
    border: 1px solid #BFBFBF;
    height: 60px;
    border-radius: 4px;
    padding: 20px;
    outline: #BFBFBF;
    transition:var(--transition);
    width:100%;
}
.form__input:focus ,.form__input:focus-visible{
    border:1px solid var(--main-color);
}
.form__input-password{
    position: relative;
}
.form__view-pass{
    --size:20px;
    position: absolute;
    width: var(--size);
    height: var(--size);
    top:calc(50% - var(--size)/2);
    right: 30px;
    z-index: 100;
    cursor: pointer;
    background-color:transparent;
    border:0;
    line-height: 0;
}
.form__view-pass-icon{
    width: var(--size);
    height: var(--size);
}
.form__view-pass-icon path{
    transition: var(--transition);
}
.toggle-view-pass{
    transform: translate(-22px, -22px);
    transition: var(--transition);
}
.login__remeber-me{
    display: flex;
    align-items: center;
    margin-bottom: 30px;
}
.login__remeber-me-cont{
    display: flex;
    align-items: center;
}
.login__remeber-me-check{
    border-radius: 2px;
    width: 15px;
    height: 15px;
    margin-right: 8px;
    accent-color: var(--main-color);
}
.login__remeber-me-label{
    font-family: 'PoppinsMedium';
    color:var(--text-color-dark);
    margin: 0;
}
.login__forget-password{
    margin-left: auto;
    font-size: 16px;
    font-family: 'PoppinsMedium';
    color: var(--main-color);
    transition: var(--main-color);
}
.login__forget-password:hover{
    color: var(--main-color);
    opacity: .8
}
.form__submit{
    display: flex;
    align-items: center;
    justify-content: center;
    width: 60%;
    height: 48px;
    border-radius: 4px;
    font-size: 20px;
    font-family: 'PoppinsMedium';
    color: white;
    background-color: var(--main-color);
    margin: auto;
    margin-bottom: 25px;
}
.login__with-other-ways{
    position: relative;
    font-size: 18px;
    color: var(--main-color);
    font-family: 'PoppinsMedium';
    text-align: center;
    margin-top: 30px;
}
.login__with-other-ways::before ,.login__with-other-ways::after{
    content:'';
    position:absolute;
    background-color: #BFBFBF;
    height:1px;
    width:35%;
    top:calc(50% - 1px/2);
}
.login__with-other-ways::before{
    left: 0;
} 
.login__with-other-ways::after{
    right: 0;
}
.login__social-media-cont{
    display: flex;
    align-items:center;
    justify-content:center;
}
.login__social-media-button{
    display: flex;
    align-items:center;
    justify-content: center;
    background-color:transparent;
    border:0;
    outline:0;
    cursor:pointer;
    margin-right: 30px;
}
.login__social-media-button:last-of-type{
    margin-right:0;
}
.login__dont-have-accout{
    text-align: center;
    color: #8B8B8B;
    font-size: 22px;
    font-family: 'poppinsMedium';
}
.login__move-to-register{
    color: var(--main-color);
    font-size: 24px;
    font-family:'PoppinsSemiBold';
    transition: var(--transition);
}
.sign-up__move-to-login:hover{
    color:var(--main-color);
    text-decoration: underline;
}
.modal__form-input-error-message{
    color: red;
    font-size: 14px;
    margin-top: 4px;
}
.modal__form-input--error{
    border-color:red!important
}
.modal__form-input--error::placeholder{
    color: red;
    font-weight: 0;
    font-size:14px
}
@media(max-width:1024px){
    .login-form-content{
        width:80%
    }
    .login__form-content-container{
        width:80%
    }
}
@media(max-width:768px){
    .login-form-content{
        width:90%
    }
    .login__form-content-container{
        width:90%
    }
    .login__with-other-ways::before, .login__with-other-ways::after{
        width: 30%  ;
    }
    #login{
        padding: 30px 5px;
    }
}
@media(max-width:426px){
    .login__form-content-container{
        width:95%
    }
    .login__remeber-me{
        flex-direction: column;
        align-items: flex-start;
    }
    .login__forget-password{
        margin-left:0
    }
    .form__submit{
        margin-bottom: 30px;
        height: 50px;
    }
    .login__with-other-ways{
        font-size: 16px;
        margin-top: 40px;
    }
    .login__remeber-me{
        margin-bottom: 20px;
    }
    .login__remeber-me-cont{
        margin-bottom: 10px;
    }
    .login__remeber-me-check{
        width: 20px;
        height: 20px;
        margin-right: 6px;
    }
    .login__alredy-have-accout{
        font-size: 14px;
    }
    .login__move-to-register{
        font-size: 16px;
    }
    .login-form-content{
        padding: 20px 10px;
        margin-bottom: 5px;
    }
    #login {
        padding: 10px 0px;
    }
    .login-form-title {
        font-size: 22px;
    }
    .form__input-label{
        margin-bottom: 5px;
    }
    .form__input {
        padding: 15px 20px;
    }
    .login__return-span{
        display:none
    }
    .login__return-icon {
        width: 40px;
        height: 40px;
        color: var(--main-color);
        margin-left: 10px;
    }
    .form__input-cont{
        margin-bottom: 15px;
    }
    .login__social-media-button{
        margin-right: 0px;
    }
    .login__logo{
        width:70px;
        height:70px;
    }
}