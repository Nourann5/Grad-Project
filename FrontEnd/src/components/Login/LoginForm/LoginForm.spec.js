import {render , screen} from '@testing-library/react'
import { Provider } from 'react-redux'
import { BrowserRouter, Route } from 'react-router-dom'
import store from 'store/store'
import LoginForm from './LoginForm'

describe('Test the Login Form', () => { 
    it('Heading exists',()=>{
        const tree =  render(
            <BrowserRouter>
                <Provider store={store}>
                    <LoginForm />
                </Provider>
            </BrowserRouter>
        )
        // const headerTitle = screen.getByRole('paragraph',{name:'Welcome Back'})
        // expect(headerTitle).toBeInTheDocument()
        expect(tree).toMatchSnapshot()
    })
})