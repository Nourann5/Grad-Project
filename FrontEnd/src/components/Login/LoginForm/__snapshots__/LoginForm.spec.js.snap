// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Login Form Heading exists 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="login-form-content"
      >
        <div
          class="login__form-content-container"
        >
          <p
            class="login-form-title"
          >
            Welcome Back
          </p>
          <form
            class="login-form"
          >
            <div
              class="form__input-cont"
            >
              <label
                class="form__input-label"
                for="form__input-email"
              >
                Email
              </label>
              <input
                class="form__input "
                id="form__input-email"
                name="email"
                placeholder="Type Email Here"
                type="email"
              />
            </div>
            <div
              class="form__input-cont"
            >
              <label
                class="form__input-label"
                for="form__input-password"
              >
                Password
              </label>
              <div
                class="form__input-password"
              >
                <input
                  class="form__input "
                  id="form__input-password"
                  name="password"
                  placeholder="Type Password Here"
                  type="password"
                />
                <button
                  class="form__view-pass"
                  type="button"
                >
                  <svg
                    class="form__view-pass-icon"
                  >
                    passwordViewed.svg
                  </svg>
                </button>
              </div>
            </div>
            <div
              class="login__remeber-me"
            >
              <div
                class="login__remeber-me-cont"
              >
                <input
                  class="login__remeber-me-check"
                  id="login__remeber-me-check"
                  type="checkbox"
                />
                <label
                  class="login__remeber-me-label"
                  for="login__remeber-me-check"
                >
                  Remember Me
                </label>
              </div>
              <a
                class="login__forget-password"
                href="/forget"
              >
                Forgot Password?
              </a>
            </div>
            <button
              class="form__submit"
              type="submit"
            >
              Login
            </button>
          </form>
          <p
            class="login__with-other-ways"
          >
            Or Login With
          </p>
          <div
            class="login__social-media-cont"
          >
            <button
              class="login__social-media-button"
            >
              <svg>
                facebook.svg
              </svg>
            </button>
            <button
              class="login__social-media-button"
            >
              <svg>
                google.svg
              </svg>
            </button>
            <button
              class="login__social-media-button"
            >
              <svg>
                linkedin.svg
              </svg>
            </button>
          </div>
        </div>
      </div>
      <p
        class="login__dont-have-accout"
      >
        Don't Have An Account ?
        <a
          class="login__move-to-register"
          href="/register"
        >
           Register
        </a>
      </p>
    </div>
  </body>,
  "container": <div>
    <div
      class="login-form-content"
    >
      <div
        class="login__form-content-container"
      >
        <p
          class="login-form-title"
        >
          Welcome Back
        </p>
        <form
          class="login-form"
        >
          <div
            class="form__input-cont"
          >
            <label
              class="form__input-label"
              for="form__input-email"
            >
              Email
            </label>
            <input
              class="form__input "
              id="form__input-email"
              name="email"
              placeholder="Type Email Here"
              type="email"
            />
          </div>
          <div
            class="form__input-cont"
          >
            <label
              class="form__input-label"
              for="form__input-password"
            >
              Password
            </label>
            <div
              class="form__input-password"
            >
              <input
                class="form__input "
                id="form__input-password"
                name="password"
                placeholder="Type Password Here"
                type="password"
              />
              <button
                class="form__view-pass"
                type="button"
              >
                <svg
                  class="form__view-pass-icon"
                >
                  passwordViewed.svg
                </svg>
              </button>
            </div>
          </div>
          <div
            class="login__remeber-me"
          >
            <div
              class="login__remeber-me-cont"
            >
              <input
                class="login__remeber-me-check"
                id="login__remeber-me-check"
                type="checkbox"
              />
              <label
                class="login__remeber-me-label"
                for="login__remeber-me-check"
              >
                Remember Me
              </label>
            </div>
            <a
              class="login__forget-password"
              href="/forget"
            >
              Forgot Password?
            </a>
          </div>
          <button
            class="form__submit"
            type="submit"
          >
            Login
          </button>
        </form>
        <p
          class="login__with-other-ways"
        >
          Or Login With
        </p>
        <div
          class="login__social-media-cont"
        >
          <button
            class="login__social-media-button"
          >
            <svg>
              facebook.svg
            </svg>
          </button>
          <button
            class="login__social-media-button"
          >
            <svg>
              google.svg
            </svg>
          </button>
          <button
            class="login__social-media-button"
          >
            <svg>
              linkedin.svg
            </svg>
          </button>
        </div>
      </div>
    </div>
    <p
      class="login__dont-have-accout"
    >
      Don't Have An Account ?
      <a
        class="login__move-to-register"
        href="/register"
      >
         Register
      </a>
    </p>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
